windows/RNCConfig/generate-header.js

```diff
@@ -0,0 +1,73 @@
+const fs = require('fs')
+const path = require('path')
+const readline = require('readline')
+
+const envFile = path.join(process.argv[2], '..', process.env['ENVFILE'] || '.env')
+const outDir = path.join(process.argv[3], 'Generated Files')
+
+console.log(`Generating files in ${outDir} from ${envFile} env file`)
+
+if (!fs.existsSync(outDir)) {
+  fs.mkdirSync(outDir)
+}
+
+if (fs.existsSync(envFile)) {
+  const vars = []
+  const regex = /^\s*(?:export\s+|)([\w\d\.\-_]+)\s*=\s*['"]?(.*?)?['"]?\s*$/
+  readline.createInterface({
+    input: fs.createReadStream(envFile),
+    console: false
+  }).on('line', (line)=>{
+    const matches = line.match(regex)
+    if (matches)
+      vars.push({key: matches[1], value: matches[2]})
+  }).on('close', ()=> {
+    generateFiles(vars)
+  });
+} else {
+  console.warn(`waring: env file ${envFile} does not exit`)
+  generateFiles([])
+}
+
+function generateFiles(vars) {
+  // Native code
+  let nativeCode = '';
+  // React Native Module code
+  let rnCode = '';
+  nativeCode += '#include<string>\n'
+  nativeCode += 'namespace ReactNativeConfig {\n'
+  for (let {key, value} of vars) {
+    const escaped = escapeString(value)
+    nativeCode += `  inline static std::string ${key} = ${escaped};\n`
+    rnCode += `REACT_CONSTANT(${key});\n`
+    rnCode += `static inline const std::string ${key} = ${escaped};\n`;
```

Comment:
these could be made `constexpr` instead and get rid of the nativeCode?

Reply:
+1 on `constexpr`

I did two files to have a simpler namespace for the values in native code - i.e. `ReactNativeConfig::something` instead of `RNCConfig::ReactNativeConfigModule::something`